maven-resolver-provider/src/main/java/org/apache/maven/repository/internal/MavenResolverModule.java

```diff
@@ -46,6 +47,7 @@
     protected void configure()
     {
         install( new AetherModule() );
+        bind( VersionScheme.class ).toProvider( new DefaultVersionSchemeProvider() );
```

Comment:
Just a quick question : is the explicit registration required ? Shouldn't sisu automatically wire the provider ?

Reply:
This class is NOT used in Maven, this is Guice module we provide for 3rd parties that use Guice for DI (so they can simply install this module and have "maven embedded"), as Resolver (the `AetherModule` above) itself is _incomplete_, Resolver has no notion of "maven models", have no idea how to read POM, repository metadata XML etc. This module "merely completes" resolver by adding "maven bits".

Sisu will pick up the DefaultVersionSchemeProvider from sisu-index, as Sisu Mojo generates index for this module.

Note: to use Guice modules within Sisu, you'd need a) directly add is somewhere (we do not), or b) annotate it with `@Named` (which is not) to have Sisu pick it up.

Also, see https://github.com/apache/maven/pull/479 and related issue.